src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +52,74 @@ public List<IexLastTradedPrice> getLastTradedPriceForSymbols(final List<String>
     }
   }
 
-
+  /**
+   * Get the historical traded prices for a given range or date.
+   *
+   * @param symbols the ticker for the stock
+   * @param ranges the time series for the historical traded price (max,5y,2y,1y,ytd,6m,3m,1m,5d)
+   * @param date the specific date (YYYYMMDD)
+   * @return A list of historical traded price objects for each Symbol that is passed in
+   */
+  public List<List<IexHistoricalPrice>> getHistoricalPriceForSymbol(
+          final List<String> symbols, final List<String> ranges, final String date) {
+    if (CollectionUtils.isEmpty(symbols)) { // symbols must not be empty
+      log.warn("User passed empty list to /HistorcalTradePrices endpoint. "
+              + "Symbols must not be empty.");
+      return Collections.emptyList();
+    } else {
+      List<List<IexHistoricalPrice>> historicalPriceListForSymbols = new ArrayList<>();
+      List<IexHistoricalPrice> iexHistoricalPrice = new ArrayList<>();
+      if (!CollectionUtils.isEmpty(ranges)) { // symbols & ranges
+        if (symbols.size() != ranges.size()) { // symbols & ranges must be the same size
+          log.warn("User passed different sized list to /HistorcalTradePrices endpoint. "
+                  + "Symbols and ranges must be the same size.");
+          return Collections.emptyList();
+        }
+        for (int i = 0; i < symbols.size(); i++) {
+          try {
+            iexHistoricalPrice = iexCloudClient
+                    .getHistoricalPricesForSymbolWithRange(symbols.get(i), ranges.get(i));
```

Comment:
This is a bit confusing, as a consumer of your API I wouldn't expect it to handle a list of symbols alongside a list of ranges since there's no ordering concept in API params. Maybe we should just limit it to one range per API call.